### ISSUE TITLE
gardener-metrics-exporter fails with msg="open : no such file or directory"

### ISSUE THREAD

**Issue by alcasim** [gardener/gardener-metrics-exporter#2]
Latest version of the exporter fails with the error:
time="2018-10-04T13:20:03Z" level=error msg="open : no such file or directory"

I'm using the included helm chart to deploy and I got success deploying with the previous version, but with the latest gardener-metrics-exporter refuses to start complaining about the aforementioned error.

I did a quick test locally and I think it expects a kubeconfig file to be specified. I think it's unnecessary to specify this for a pod which is "usually" intended to be run in the same namespace than gardener-controller-manager pod (kubeconfig option is useful for debug purposes, although), or maybe I'm getting wrong the purpose of the kubeconfig file.

How should be the deployment done in the latest version? I'm happy to help contributing on fixing the helm chart

**Comment by dkistner** [comment]
Did a short test based on latest master commit: 

I used this values.yaml with no kubeconfig specified (then a ServiceAccount and corresponding ClusterRole/ClusterRoleBinding should be deployed).
```yaml
image:
  tag: 0.0.0
```

Then I run the following:
```bash
# move to repo and deploy values.yaml
cd $GOPATH/src/github.com/gardener/gardener-metrics-exporter
cat <<EOF > dev/values.yaml
image:
  tag: 0.0.0
EOF

# deploy test ns
$ kubectl create ns test >/dev/null

# deploy helm chart
$ helm upgrade gardener-metrics-exporter $GOPATH/src/github.com/gardener/gardener-metrics-exporter/charts/gardener-metrics-exporter --install --force --wait --values dev/values.yaml --namespace test
Release "gardener-metrics-exporter" does not exist. Installing it now.
NAME:   gardener-metrics-exporter
LAST DEPLOYED: Fri Oct  5 16:28:46 2018
NAMESPACE: test
STATUS: DEPLOYED

RESOURCES:
==> v1/Service
NAME                       TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)   AGE
gardener-metrics-exporter  ClusterIP  10.110.164.247  <none>       2718/TCP  2s

==> v1/Deployment
NAME                       DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
gardener-metrics-exporter  1        1        1           1          2s

==> v1/Pod(related)
NAME                                        READY  STATUS       RESTARTS  AGE
gardener-metrics-exporter-5ddb5d5f7c-9g5c6  1/1    Running      0         2s

==> v1/ServiceAccount
NAME                       SECRETS  AGE
gardener-metrics-exporter  1        2s

==> v1/ClusterRole
NAME                                 AGE
garden.sapcloud.io:metrics-exporter  2s

==> v1/ClusterRoleBinding
NAME                                 AGE
garden.sapcloud.io:metrics-exporter  2s
```

Now verify if it is running
```bash
$ k -n test get pod
NAME                                         READY     STATUS    RESTARTS   AGE
gardener-metrics-exporter-5ddb5d5f7c-9g5c6   1/1       Running   0          35s
```

Be aware there is not helm chart repo for the exporter (it is located in the repo) therefore you have to pass an absolute path to the chart location to ``helm upgrade/install`` like I did upfront (`$GOPATH/src/github.com/gardener/gardener-metrics-exporter/charts/gardener-metrics-exporter`).

**Comment by alcasim** [comment]
Hey, thanks @dkistner for your response, I'm currently using image value "0.1.0-dev-8fde706255c44c8f39aeb76ab0563f953e485c40" as it's the latest image before introducing the kubeconfig path flag

In my use case, I'm running the exporter in same namespace and cluster than gardener-controller-manager, I agree on having this flag is useful to run the exporter from a different kubernetes cluster

May I suggest something? Perhaps the default exporter behavior should be relying on the ServiceAccount used to run the pod (as it was in the beginning) so running the exporter on same cluster and namespace would be supported out of the box, or use the kubeconfig if it has been specified with the flag, that would cover all use cases (running together with the gardener-controller-manager, and running outside namespace/cluster)

**Comment by dkistner** [comment]
Yes you are absolutely right, we should do that in the way you described. I will address that.

**Comment by dkistner** [comment]
Fixed 217ac9f65228bd1efc140021200fef0c40679c24
